moped-editor/src/components/GridTable/Filters.js

```diff
@@ -230,215 +129,109 @@ const Filters = ({
+      return filters;
+    } else {
+      return [generateEmptyFilter()];
+    }
+  }, [filters]);
+
   /**
    * The current local filter parameters so that we can store updated filters and
    * then apply them to the query when the search button is clicked.
    * @type {Object} filterParameters - Contains all the current filters
    * @function setFilterParameters - Update the state of filterParameters
    * @default {filters}
    */
-  const initialFilterParameters = useMemo(() => {
-    if (filters.length > 0) {
-      return makeInitialFilterParameters(filters, filtersConfig);
-    } else {
-      return {
-        [uuidv4()]: generateEmptyField(uuidv4()),
-      };
-    }
-  }, [filters, filtersConfig]);
-
   const [filterParameters, setFilterParameters] = useState(
     initialFilterParameters
   );
 
   /* Track toggle value so we update the query value in handleApplyButtonClick */
   const [isOrToggleValue, setIsOrToggleValue] = useState(isOr);
 
-  /**
-   * Tracks whether the user has added a complete filter
-   */
-  const [filterComplete, setFilterComplete] = useState(false);
-
-  /* First filter is an empty placeholder so we check for more than one filter */
-  const areMoreThanOneFilters = Object.keys(filterParameters).length > 1;
-  const isFirstFilterIncomplete =
-    Object.keys(filterParameters).length === 1 && !filterComplete;
-
-  const generateEmptyFilter = useCallback(() => {
-    // Generate a random UUID string
-    const uuid = uuidv4();
-    // Clone state
-    const filtersNewState = {
-      ...filterParameters,
-    };
-    // Patch new state
-    filtersNewState[uuid] = generateEmptyField(uuid);
-    // Update new state
-    setFilterParameters(filtersNewState);
-  }, [filterParameters, setFilterParameters]);
-
-  /**
-   * Returns true if Field has a lookup table associated with it and operator is case sensitive
-   */
-  const renderAutocompleteInput = (field) => {
-    return (
-      field.lookup_table &&
-      !loading &&
-      AUTOCOMPLETE_OPERATORS.includes(field.operator)
-    );
-  };
+  /* Some features like all/any radios require more than one filter to appear */
+  const areMoreThanOneFilters = filterParameters.length > 1;
 
   /**
    * Handles the click event on the field drop-down menu
-   * @param {string} filterId - State FieldID to modify
+   * @param {string} filterIndex - filterParameters index to modify
    * @param {Object} field - The field object being clicked
    */
-  const handleFilterFieldMenuClick = (filterId, field) => {
-    // If the filter exists
-    if (filterId in filterParameters) {
-      // Clone state
-      const filtersNewState = { ...filterParameters };
-
-      // Find the field we need to gather options from
-      const fieldDetails = filtersConfig.fields.find(
-        (filter) => filter.name === field
-      );
-
-      if (!fieldDetails) {
-        filtersNewState[filterId] = generateEmptyField(filterId);
-      } else {
-        // Update field & type
```

Comment:
From here and below in this `handleFilterFieldMenuClick` function, the code was building up new state (similar to what `makeInitialFilterParameters` above was doing). This happens in the return below where `filterParameters.map` happens. The available operators logic here is captured in a new `getAvailableOperators` helper.